src/meta/MetaHttpDownloadHandler.cpp

```diff
@@ -0,0 +1,227 @@
+    localPath_ = headers->getQueryParam("localPath");
+    spaceID_ = headers->getIntQueryParam("spaceID");
+}
+
+
+void MetaHttpDownloadHandler::onBody(std::unique_ptr<folly::IOBuf>) noexcept {
+    // Do nothing, we only support GET
+}
+
+
+void MetaHttpDownloadHandler::onEOM() noexcept {
+    switch (err_) {
+        case HttpCode::E_UNSUPPORTED_METHOD:
+            ResponseBuilder(downstream_)
+                .status(405, "Method Not Allowed")
+                .sendWithEOM();
+            return;
+        case HttpCode::E_ILLEGAL_ARGUMENT:
+            ResponseBuilder(downstream_)
+                .status(400, "Bad Request")
+                .sendWithEOM();
+            return;
+        default:
+            break;
+    }
+
+    if (auto hadoopHome = std::getenv("HADOOP_HOME")) {
+        LOG(INFO) << "Hadoop Path : " << hadoopHome;
+        if (dispatchSSTFiles(hdfsUrl_, hdfsPort_, hdfsPath_, localPath_)) {
+            ResponseBuilder(downstream_)
+                .status(200, "SSTFile dispatch successfully")
+                .body("SSTFile dispatch successfully")
+                .sendWithEOM();
+        } else {
+            LOG(ERROR) << "SSTFile dispatch failed";
+            ResponseBuilder(downstream_)
+                .status(404, "SSTFile dispatch failed")
+                .sendWithEOM();
+        }
+    } else {
+        LOG(ERROR) << "Hadoop Home not exist";
+        ResponseBuilder(downstream_)
+            .status(404, "HADOOP_HOME not exist")
+            .sendWithEOM();
+    }
+}
+
+
+void MetaHttpDownloadHandler::onUpgrade(UpgradeProtocol) noexcept {
+    // Do nothing
+}
+
+
+void MetaHttpDownloadHandler::requestComplete() noexcept {
+    delete this;
+}
+
+
+void MetaHttpDownloadHandler::onError(ProxygenError error) noexcept {
+    LOG(ERROR) << "Web service MetaHttpDownloadHandler got error : "
+               << proxygen::getErrorString(error);
+}
+
+std::string MetaHttpDownloadHandler::toStr(folly::dynamic& vals) const {
+    std::stringstream ss;
+    for (auto& counter : vals) {
+        auto& val = counter["value"];
+        ss << counter["name"].asString() << "="
+           << val.asString()
+           << "\n";
+    }
+    return ss.str();
+}
+
+bool MetaHttpDownloadHandler::dispatchSSTFiles(const std::string& url,
+                                       int port,
+                                       const std::string& path,
+                                       const std::string& local) {
+    auto command = folly::stringPrintf("hdfs dfs -ls hdfs://%s:%d%s ",
+                                       url.c_str(), port, path.c_str());
+    auto result = ProcessUtils::runCommand(command.c_str());
+    std::vector<std::string> files;
+    folly::split("\n", result.value(), files, true);
+    auto partNumber = std::count_if(files.begin(), files.end(), [](const std::string& element) {
+                                        return element.find("part") != std::string::npos;
```

Comment:
I am not sure about the format stored on hdfs.  Please contact with @spacewalkman to verify it.  You could also  check it in #420

Reply:
HDFS output format looks like by default : 

```
-rw-r--r--   1 darion.wang supergroup          0 2019-06-11 10:27 hdfs://0.0.0.0:9000/data/part-00000
```

Reply:
you should double check it.  As far as i know,  the format stored on hdfs is not what you want.